goalie/mesh_seq.py

```diff
@@ -350,24 +349,42 @@ def _outputs_consistent(self):
         Assert that function spaces, initial conditions, and forms are given in a
         dictionary format with :attr:`MeshSeq.fields` as keys.
         """
-        for method in ["get_function_spaces", "get_initial_condition", "get_form"]:
-            if getattr(self, f"_{method}") is None:
+        for method in ["function_spaces", "initial_condition", "form", "solver"]:
+            if getattr(self, f"_get_{method}") is None:
                 continue
-            method_map = getattr(self, method)
-            if method == "get_function_spaces":
+            method_map = getattr(self, f"get_{method}")
+            if method == "function_spaces":
                 method_map = method_map(self.meshes[0])
-            elif method == "get_initial_condition":
+            elif method == "initial_condition":
                 method_map = method_map()
-            elif method == "get_form":
+            elif method == "form":
                 self._reinitialise_fields(self.get_initial_condition())
                 method_map = method_map()(0)
-            assert isinstance(method_map, dict), f"{method} should return a dict"
+            elif method == "solver":
+                self._reinitialise_fields(self.get_initial_condition())
+                solver_gen = method_map()(0)
+                assert hasattr(solver_gen, "__next__"), "solver should yield"
+                if logger.level == DEBUG:
+                    next(solver_gen)
+                    try:
+                        f, f_ = self.fields[next(iter(self.fields))]
+                        if (f.dat.data != f_.dat.data).all():
+                            self.debug(
+                                "Current and lagged solutions are equal. Does the"
+                                " solver yield before updating lagged solutions?"
+                            )
+                    except Exception:
+                        # TODO: extend above check to the case of mixed function spaces,
+                        # and empty self.fields (tests in test_fp_iteration.py)
```

Comment:
Any new TODOs need to reference issues so you'll need to create one and mention it here before this gets merged.

Reply:
Oh sorry, I actually meant to address this in this PR! I forgot

Reply:
Fixed in c0abbab024ffc394978cef64dfc56b464adb2132